File: MLR-spring/src/main/java/com/mlr/dto/TransactionDto.java
package com.mlr.dto;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class TransactionDto {
    private Long id;
    private Double amount;
    private String type; 
    private String description;
    private String categoryName;
    private String upiId; 
    private LocalDateTime createdAt;
}

File: MLR/DTOs/TransactionDto.cs
namespace MLR.DTOs;

public class TransactionDto
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Type { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string CategoryName { get; set; } = string.Empty;

    public string? UpiId { get; set; }
    public DateTime CreatedAt { get; set; }
}